src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,54 @@
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: #{$dayGap};
+  padding: 10px;
+  width: 7 * $calendarDaySize + 6 * $dayGap;
+
+  @for $lastDay from 28 through 31 {
+    &--month-length--#{$lastDay}
+    &__day:nth-child(n + #{$lastDay + 1}) {
+      display: none;
+    }
+  }
+
+  &__day {
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    box-sizing: border-box;
+    @include square(#{$calendarDaySize});
+    border: 1px solid black;
+    background-color: #eee;
+    transition: background-color 0.5s, transform 0.5s;
+
+    @for $number from 1 through 31 {
+      &:nth-child(#{$number})::before {
+        content: "#{$number}";
+        font-family: Arial, sans-serif;
+        font-size: 30px;
+      }
+    }
+
+    $days:
+      mon 0,
+      tue ($calendarDaySize + $dayGap),
+      wed (2 * $calendarDaySize + 2 * $dayGap),
+      thu (3 * $calendarDaySize + 3 * $dayGap),
+      fri (4 * $calendarDaySize + 4 * $dayGap),
+      sat (5 * $calendarDaySize + 5 * $dayGap),
+      sun (6 * $calendarDaySize + 6 * $dayGap);
```

Comment:
move these calculations inside loop, do not repeat yourself

Reply:
done